typescript/infra/scripts/verify-validator.ts

```diff
@@ -104,170 +89,259 @@ class S3Wrapper {
-  );
-  if (extraCheckpoints.length)
-    console.log(
-      `Extra checkpoints (${extraCheckpoints.length}): ${extraCheckpoints}\n`,
-    );
-  if (missingCheckpoints.length)
-    console.log(
-      `Missing checkpoints (${missingCheckpoints.length}): ${missingCheckpoints}\n`,
-    );
-  if (invalidCheckpoints.length)
-    console.log(
-      `Invalid checkpoints (${invalidCheckpoints.length}): ${invalidCheckpoints}\n`,
-    );
+  private async getLatestCheckpoints(): Promise<{
+    controlLatestCheckpoint: number;
+    prospectiveLastCheckpoint: number;
+  }> {
+    const [
+      { obj: controlLatestCheckpoint },
+      { obj: prospectiveLastCheckpoint },
+    ] = await Promise.all([
+      this.controlS3BucketClient
+        .getS3Obj<number>('checkpoint_latest_index.json')
+        .catch((err) => {
+          console.error(
+            "Failed to get control validator's latest checkpoint.",
+            err,
+          );
+          process.exit(1);
+        }),
+      this.prospectiveS3BucketClient
+        .getS3Obj<number>('checkpoint_latest_index.json')
+        .catch((err) => {
+          console.error(
+            "Failed to get prospective validator's latest checkpoint.",
+            err,
+          );
+          process.exit(1);
+        }),
+    ]);
 
-  if (modTimeDeltasS.length > 1) {
-    // Drop the time of the first one since it is probably way off
-    modTimeDeltasS.length--;
-    console.log(
-      `Time deltas (∆ < 0 -> prospective came earlier than the control)`,
-    );
-    console.log(modTimeDeltasS);
-    console.log(`Median: ${median(modTimeDeltasS)}s`);
-    console.log(`Mean:   ${mean(modTimeDeltasS)}s`);
-    console.log(`Stdev:  ${stdDev(modTimeDeltasS)}s`);
+    if (
+      !isLatestCheckpoint(controlLatestCheckpoint) ||
+      !isLatestCheckpoint(prospectiveLastCheckpoint)
+    )
+      throw new Error('Invalid latest checkpoint data');
+
+    console.log(`Latest Index`);
+    console.log(`control: ${controlLatestCheckpoint}`);
+    console.log(`prospective: ${prospectiveLastCheckpoint}\n`);
+    return { controlLatestCheckpoint, prospectiveLastCheckpoint };
+  }
+
+  private async getControlCheckpoint(
+    checkpointIndex: number,
+  ): Promise<{ control: Checkpoint; controlLastMod: Date } | null> {
+    let control: Checkpoint, controlLastMod: Date, unrecoverableError;
+    try {
+      const t = await this.controlS3BucketClient.getS3Obj(
+        this.checkpointKey(checkpointIndex),
+      );
+      if (isCheckpoint(t.obj)) {
+        if (t.obj.checkpoint.index != checkpointIndex) {
+          console.log(`${checkpointIndex}: Control index is invalid`, t);
+          process.exit(1);
+        }
+        [control, controlLastMod] = [t.obj, t.modified];
+      } else {
+        console.log(`${checkpointIndex}: Invalid control checkpoint`, t);
+        unrecoverableError = new Error('Invalid control checkpoint.');
+      }
+    } catch (err) {
+      return null;
+    }
```

Comment:
I don't understand what this try catch does?
why not just throw if `t.obj` is not a checkpoint and remove the `if (unrecoverableError)`

Reply:
Because the throw would be caught be the try/catch that is catching an error we want to handle. If the control checkpoint is not present but the prospective validator has one, we should record that. If we throw an error instead of catching that error we would not be able to check.